Src/Plugins/Security/SEC1004.UseSecureApi.json

```diff
@@ -2,17 +2,1907 @@
+              "Find": "{insecureApi}",
+              "ReplaceWith": "{secureApi}"
+            }
+          }
+        },
+        {
+          "SubId": "strcpy",
+          "ContentsRegex": "\\b(?<insecureApi>strcpy)\\s*\\(",
+          "MessageArguments": {
+            "secureApi": ""
+          },
+          "Fixes": {
+            "directReplacement": {
+              "Description": "Replace '{insecureApi} with '{secureApi}'.",
+              "Find": "{insecureApi}",
+              "ReplaceWith": "{secureApi}"
+            }
+          }
+        },
+        {
+          "SubId": "StrCpy",
+          "ContentsRegex": "\\b(?<insecureApi>StrCpy)\\s*\\(",
+          "MessageArguments": {
+            "secureApi": ""
+          },
+          "Fixes": {
+            "directReplacement": {
+              "Description": "Replace '{insecureApi} with '{secureApi}'.",
+              "Find": "{insecureApi}",
+              "ReplaceWith": "{secureApi}"
+            }
+          }
+        },
+        {
+          "SubId": "strncpy",
+          "ContentsRegex": "\\b(?<insecureApi>strncpy)\\s*\\(",
+          "MessageArguments": {
+            "secureApi": ""
+          },
+          "Fixes": {
+            "directReplacement": {
+              "Description": "Replace '{insecureApi} with '{secureApi}'.",
+              "Find": "{insecureApi}",
+              "ReplaceWith": "{secureApi}"
+            }
+          }
+        },
+        {
+          "SubId": "wcscpy",
+          "ContentsRegex": "\\b(?<insecureApi>wcscpy)\\s*\\(",
+          "MessageArguments": {
+            "secureApi": ""
+          },
+          "Fixes": {
+            "directReplacement": {
+              "Description": "Replace '{insecureApi} with '{secureApi}'.",
+              "Find": "{insecureApi}",
+              "ReplaceWith": "{secureApi}"
+            }
+          }
+        },
+        {
+          "SubId": "wcsncpy",
+          "ContentsRegex": "\\b(?<insecureApi>wcsncpy)\\s*\\(",
+          "MessageArguments": {
+            "secureApi": ""
+          },
+          "Fixes": {
+            "directReplacement": {
+              "Description": "Replace '{insecureApi} with '{secureApi}'.",
+              "Find": "{insecureApi}",
+              "ReplaceWith": "{secureApi}"
+            }
+          }
+        }
+      ]
+    },
+    {
+      "Id": "SEC1004/String/Concatenation",
+      "Name": "DoNotUseBannedApi",
+      "Description": "Developers should use secure API in preferance of insecure alternates.",
+      "Message": "'{0:scanTarget}' contains a call to '{1:insecureApi}', a potentially insecure API that could be replaced with a more secure alternative: '{2:secureApi}'.",
+      "Level": "Error",
+      "FileNameAllowRegex": "$CSourceFiles",
+      "MatchExpressions": [
+        {
+          "SubId": "_mbscat",
+          "ContentsRegex": "\\b(?<insecureApi>_mbscat)\\s*\\(",
+          "MessageArguments": {
+            "secureApi": ""
```

Comment:
>"secureApi": "" [](start = 12, length = 15)

use a comma separated list of API here. so that it reads, with a more secure alternative: 'a, b or c'